lib/AST/Evaluator.cpp

```diff
@@ -378,24 +379,93 @@ void Evaluator::dumpDependenciesGraphviz() const {
   printDependenciesGraphviz(llvm::dbgs());
 }
 
+void evaluator::DependencyCollector::realizeOrRecord(const Reference &ref) {
+  if (isRecording) {
+    scratch.insert(std::move(ref));
+    return;
+  }
+
+  auto *tracker = getActiveDependencyTracker();
+  if (!tracker) {
+    return;
+  }
+
+  switch (ref.kind) {
+  case Reference::Kind::Empty:
+  case Reference::Kind::Tombstone:
+    llvm_unreachable("cannot record empty dependency");
+  case Reference::Kind::UsedMember:
+    tracker->addUsedMember({ref.subject, ref.name}, isActiveSourceCascading());
```

Comment:
isActiveSourceCascading() can go away entirely right?

Reply:
The legacy infrastructure still goes through this code too, so unfortunately not yet 😞